include/swift/Basic/SourceManager.h

```diff
@@ -32,8 +32,17 @@ class GeneratedSourceInfo {
 public:
   /// The kind of generated source code.
   enum Kind {
-    /// The expansion of a macro.
-    MacroExpansion,
+    /// The expansion of a freestanding expression macro.
+    ExpressionMacroExpansion,
```

Comment:
I was thinking of metaprogramming the definition of `MacroRole`, since we need a bunch of little information about it. I can do that down in `include/swift/Basic/MacroRoleKinds.def` so we can generate these rather than mirror them.